File: Android - Native/discover_destination/app/src/main/java/com/example/discover_destination/viewmodel/UsersViewModel.kt
package com.example.discover_destination.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.discover_destination.domain.User
import com.example.discover_destination.uscases.UsersUseCases
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class UsersViewModel @Inject constructor(
    private val useCase: UsersUseCases
) : ViewModel(){

    init {
        viewModelScope.launch {
            useCase()
        }
    }

    fun checkUser(user: User): Boolean {
        return useCase.invokeCheckUser(user)
    }
}

File: Flutter - Non-Native/discover_destination/lib/viewmodel/users_view_model.dart
import 'package:discover_destination/model/user.dart';
import 'package:discover_destination/repository/users_repository.dart';

class UsersViewModel{
  UsersRepository repository;

  UsersViewModel(this.repository);

  bool checkUser(User user) {
    return repository.checkUser(user);
  }
}